### PR TITLE
HHH-18528 Test strict Jaxp security settings with JDK 23 / HHH-18488 CI check for build reproducibility

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HHH-18528
https://hibernate.atlassian.net/browse/HHH-18488

- build reproducibility check builds on top of the previous PR and executes the check on CI. It will build ORM two times publishing artifacts to two different "maven repositories" and then use the `compare-build-results.sh` script to compare the artifacts. The script returns !=0 if there are some differences (so that should fail the build). 
- As for the JAXP strict settings, I've done the same as for the Validator check, but if anyone has other ideas I'm happy to try them out 😃 

And the build with these two new executions is running here: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-orm-personal-marko/detail/build%2Frepo-check/30/pipeline/

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

File: nightly.Jenkinsfile [hunk 1]
```diff
@@ -204,6 +204,68 @@ stage('Build') {
 			}
 		})
 	}
+	// Don't run additional checks when this is a PR
```
Comment (yrodiere) [inline nightly.Jenkinsfile:207]: We run the nightly on PRs? :thinking:

File: gradle/java-module.gradle [hunk 2]
```diff
@@ -228,17 +228,19 @@ tasks.withType( Test.class ).each { test ->
...
-	test.jvmArgs += [
-			'-XX:+HeapDumpOnOutOfMemoryError',
-			'-XX:MetaspaceSize=256M'
-	]
+	// Byteman needs this property to be set, https://developer.jboss.org/thread/274997
+	test.jvmArgs('-Djdk.attach.allowAttachSelf=true')
+
+	if (project.hasProperty('additionalTestJvmArg')) {
+		test.jvmArgs(project.property('additionalTestJvmArg'))
+	}
```
Comment (yrodiere) [inline gradle/java-module.gradle:240]: I think this is already supported with `-Ptest.jdk.launcher.args=...`.

See https://github.com/hibernate/hibernate-orm/blob/e407c53c861a0e9dc745a78b15a5b1afd9ffd7c7/Jenkinsfile#L127C13-L127C35

File: nightly.Jenkinsfile (hunk 1 above)
Comment (marko-bekhta) [reply nightly.Jenkinsfile:207]: I hope not.. but then there's this

https://github.com/hibernate/hibernate-orm/blob/e407c53c861a0e9dc745a78b15a5b1afd9ffd7c7/nightly.Jenkinsfile#L77-L80

so 🤷🏻 I thought I'd better put that if

File: gradle/java-module.gradle (hunk 2 above)
Comment (marko-bekhta) [reply gradle/java-module.gradle:240]: ohh ok, I didn't see anything in gradle file and wasn't sure, but it seems this is handled here:

https://github.com/hibernate/hibernate-orm/blob/e407c53c861a0e9dc745a78b15a5b1afd9ffd7c7/local-build-plugins/src/main/java/org/hibernate/orm/toolchains/JavaModulePlugin.java#L128-L134

will make an update ...
Thanks 😃

Comment (sebersole) [reply gradle/java-module.gradle]: > ohh ok, I didn't see anything in gradle file

We try to move as much as possible to (local) plugins to keep the scripts as simple as possible.

**Comment by sebersole** [comment]
Thanks!

**Pull request merged**